app/Models/User.php

```diff
@@ -869,25 +863,64 @@ public function sumOfActiveFreePages(): int
     /* Transaction related */
 
     /**
-     * Returns the payed kkt amount in the semester. 0 if has not payed kkt.
+     * Pay kkt and netreg.
+     * Also updates workshop balances
+     * and the internet access expiry date.
+     * Returns an array with the two transaction objects
+     * and the new expiry date.
+     */
+    public function payKKTNetreg(int $kkt_amount, int $netreg_amount)
+    {
+        // Creating transactions
+        $kkt = Transaction::create([
+            'checkout_id' => Checkout::studentsCouncil()->id,
+            'receiver_id' => Checkout::studentsCouncil()->handler_id,
+            'payer_id' => $this->id,
+            'semester_id' => Semester::current()->id,
+            'amount' => $kkt_amount,
+            'payment_type_id' => PaymentType::kkt()->id,
+            'comment' => null,
+            'moved_to_checkout' => null,
+        ]);
+
+        $netreg = Transaction::create([
+            'checkout_id' => Checkout::admin()->id,
+            'receiver_id' => Checkout::studentsCouncil()->handler_id, // beware; the student council collects this, too!
+            'payer_id' => $this->id,
+            'semester_id' => Semester::current()->id,
+            'amount' => $netreg_amount,
+            'payment_type_id' => PaymentType::netreg()->id,
+            'comment' => null,
+            'moved_to_checkout' => null,
+        ]);
+
+        WorkshopBalance::generateBalances(Semester::current()->id);
+
+        $new_expiry_date = \App\Http\Controllers\Network\InternetController::extendUsersInternetAccess($this);
+
+        return [$kkt, $netreg, $new_expiry_date];
+    }
+
+    /**
+     * Returns the payed kkt amount in the semester; or null if the user has not payed kkt.
      * @param Semester $semester
-     * @return int
+     * @return ?int
      */
-    public function payedKKTInSemester(Semester $semester): int
+    public function payedKKTInSemester(Semester $semester): ?int
```

Comment:
Would you rename everything to "paidSomething" while you are here, please?

Reply:
I don't understand this here; should I generalise the function to be able to pay anything? Now it is specially about kkt.

Reply:
Oh, I see:)